### PR TITLE
Introducing new NotificationCenter

### PR DESCRIPTION
**Pull request by peuter**
this PR adds a central NotificationRouter, which allows to route messages by topic to different end-points (like Popups, or the newly added NotificationCenter).
Some built-in routes were added too, those are:

* a GlobalErrorHandler catches all uncatched exceptions and generates an error notification, which is shown as a popup with an action-button to open an issue on github with some usefull information (Note: the GlobalErrorhandler is only activated in build not in source version)
* a message is shown to inform the user when the backend connection is gone
* User can have customized messages for incoming state updates (definition be added to the config) e.g. to get a notification when an motion sensor gets activated

This is the first draft I am about to add more features to it, but as the changeset is quite huge already I want to merge this first and add an enhancement issue for the remaining features.

### CODE REVIEW

**Comment by peuter** [comment]
btw. the other design should work too now.

File: source/class/cv/core/notifications/actions/AbstractActionHandler.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* NotificationCenter.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/actions/AbstractActionHandler.js]: Here and in other places: Copyright info should state 2017

File: source/class/cv/parser/widgets/NotificationCenterBadge.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/* Text.js
```
Comment (ChristianMayer) [inline source/class/cv/parser/widgets/NotificationCenterBadge.js]: Wrong name

File: source/class/cv/plugins/openhab/Openhab.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+/* cometvisu.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/plugins/openhab/Openhab.js]: wrong name

File: source/class/cv/ui/BodyBlocker.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/* BodyBlocker.js
```
Comment (ChristianMayer) [inline source/class/cv/ui/BodyBlocker.js]: What is the purpose of this method?

File: source/class/cv/ui/Popup.js [hunk 5]
```diff
@@ -77,37 +77,55 @@ qx.Class.define('cv.ui.Popup', {
...
-        var content = qx.bom.Html.clean(['<div class="main"></div>'])[0];
+        var content = qx.dom.Element.create("div", { "class": "main"});
         qx.dom.Element.insertEnd(content, ret_val);
         if (qx.lang.Type.isString(attributes.content)) {
-          var mainContent =  qx.bom.Html.clean([attributes.content])[0];
-          qx.dom.Element.insertEnd(mainContent, content);
+          var html = ""+attributes.content;
+          if (attributes.icon) {
+            var icon = qx.util.ResourceManager.getInstance().toUri("icon/knx-uf-iconset.svg")+"#kuf-"+attributes.icon;
+            console.log(icon);
```
Comment (ChristianMayer) [inline source/class/cv/ui/Popup.js]: Left over debug message?

File: source/class/cv/ui/structure/pure/NotificationCenterBadge.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
+/* Text.js
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/pure/NotificationCenterBadge.js]: File name

File: source/class/cv/core/notifications/ActionRegistry.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+/* NotificationCenter.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/ActionRegistry.js]: Name

File: source/class/cv/core/notifications/IActionHandler.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+/* INotificationHandler.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/IActionHandler.js]: Name

File: source/class/cv/core/notifications/IHandler.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+/* INotificationHandler.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/IHandler.js]: Name

File: source/class/cv/core/notifications/Router.js [hunk 10]
```diff
@@ -0,0 +1,261 @@
+/* NotificationCenter.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/Router.js]: Name

File: source/class/cv/core/notifications/actions/AbstractActionHandler.js (hunk 1 above)
Comment (ChristianMayer) [inline source/class/cv/core/notifications/actions/AbstractActionHandler.js]: Name

File: source/class/cv/core/notifications/actions/Link.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
+/* NotificationCenter.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
```
Comment (ChristianMayer) [inline source/class/cv/core/notifications/actions/Link.js]: Name

File: source/class/cv/ui/BodyBlocker.js (hunk 4 above)
Comment (peuter) [reply source/class/cv/ui/BodyBlocker.js]: Qooxdoo provides Blockers that overlay certain elements, e.g. add an opacity to the background while you show an popup. The blocker adds the overlay and catches all events to disable any usage of elements in background while the popup is visible.

**Comment by ChristianMayer** [comment]
closes #603

**Pull request merged**